azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -539,6 +549,50 @@ public int selectAndUpdateExecutionWithLocking(final int executorId, final boole
     }
   }
 
+  private static class FetchAgedQueuedExecutableFlows implements
+      ResultSetHandler<List<Pair<ExecutionReference, ExecutableFlow>>> {
+
+    // Select queued unassigned flows
+    private static final String FETCH_FLOWS_QUEUED_FOR_LONG_TIME =
+        "SELECT exec_id, enc_type, flow_data, status FROM execution_flows"
+            + " WHERE submit_time < ? AND status = "
+            + Status.PREPARING.getNumVal();
+
+    @Override
+    public List<Pair<ExecutionReference, ExecutableFlow>> handle(final ResultSet rs)
+        throws SQLException {
+      if (!rs.next()) {
+        return Collections.emptyList();
+      }
+
+      final List<Pair<ExecutionReference, ExecutableFlow>> execFlows =
+          new ArrayList<>();
+      do {
+        final int id = rs.getInt(1);
+        final int encodingType = rs.getInt(2);
+        final byte[] data = rs.getBytes(3);
+
+        if (data == null) {
+          ExecutionFlowDao.logger.error("Found a flow with empty data blob exec_id: " + id);
+        } else {
+          final EncodingType encType = EncodingType.fromInteger(encodingType);
+          final Status status = Status.fromInteger(rs.getInt(4));
+          try {
+            final ExecutableFlow exFlow =
+                ExecutableFlow.createExecutableFlow(
+                    GZIPUtils.transformBytesToObject(data, encType), status);
+            final ExecutionReference ref = new ExecutionReference(id);
+            execFlows.add(new Pair<>(ref, exFlow));
+          } catch (final IOException e) {
+            throw new SQLException("Error retrieving flow data " + id, e);
+          }
+        }
+      } while (rs.next());
+
```

Comment:
This seems to have significant duplication with `FetchQueuedExecutableFlows`. Could you please see if the existing class can be re-used with an additional query/query-parameter ?

Reply:
Done. Merged it with FetchExecutableFlows.

Reply:
Has this change been committed? I can't find the commit.

Reply:
@anish-mal how was this resolved? I am unable to see the change either. 
Perhaps the commit was not pushed ?